open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/data-engine-connectors/data-engine-proxy-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/dataengineproxy/DataEngineConnectorBase.java

```diff
@@ -2,27 +2,182 @@
 /* Copyright Contributors to the ODPi Egeria project. */
 package org.odpi.openmetadata.openconnectors.governancedaemonconnectors.dataengineproxy;
 
+import org.odpi.openmetadata.accessservices.dataengine.client.DataEngineImpl;
 import org.odpi.openmetadata.accessservices.dataengine.model.LineageMapping;
 import org.odpi.openmetadata.accessservices.dataengine.model.Process;
+import org.odpi.openmetadata.accessservices.dataengine.model.SoftwareServerCapability;
 import org.odpi.openmetadata.frameworks.connectors.ConnectorBase;
+import org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException;
+import org.odpi.openmetadata.frameworks.connectors.ffdc.OCFRuntimeException;
+import org.odpi.openmetadata.frameworks.connectors.ffdc.PropertyServerException;
+import org.odpi.openmetadata.frameworks.connectors.ffdc.UserNotAuthorizedException;
+import org.odpi.openmetadata.frameworks.connectors.properties.ConnectionProperties;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.List;
+import java.util.Map;
 
 public class DataEngineConnectorBase extends ConnectorBase implements DataEngineInterface {
 
     private static final Logger log = LoggerFactory.getLogger(DataEngineConnectorBase.class);
 
+    private DataEngineImpl dataEngineClient;
+
     public DataEngineConnectorBase() { super(); }
 
-    // TODO: provide base implementation of interface methods once Data Engine OMAS client is available
-    public void sendProcess(Process process) {
-        log.debug("Placeholder for sending a process in base DataEngineConnector implementation.");
+    /**
+     * Call made by the ConnectorProvider to initialize the Connector with the base services.
+     *
+     * @param connectorInstanceId   unique id for the connector instance useful for messages etc
+     * @param connectionProperties   POJO for the configuration used to create the connector.
+     */
+    @Override
+    public void initialize(String               connectorInstanceId,
+                           ConnectionProperties connectionProperties) {
+        super.initialize(connectorInstanceId, connectionProperties);
+        Map<String, Object> omasProperties = this.connectionBean.getSecuredProperties();
+        if (omasProperties != null) {
+            // Setup the Data Engine OMAS connection based on the secured client sent through from the connector's
+            // configuration and initialization (from DataEngineProxyOperationalServices)
+            dataEngineClient = (DataEngineImpl) omasProperties.getOrDefault("dataEngineOMASClient", null);
+        }
```

Comment:
In our initial example (DataStage) there is no event, and technically not even an API (we're wrangling another API to get what we're after) -- but that's probably just detail.

I guess what we're after is that if there is no event that the data engine proxy server itself does the periodic polling?  Does that mean it also retains state on when its last successful poll completed?  (At the moment I'm handling state persistence through the data engine connector itself: leaving it to the data engine itself to manage the state of what's been synced)...

Reply:
Agreed - this is the passive open technology pattern where polling is required

![technology-integration-patterns-passive-open-technology](https://user-images.githubusercontent.com/12354195/60806727-ae5eab80-a17b-11e9-86b2-b73a14cfe994.png)

Reply:
Are you assuming that each data engine server proxy is serving one or many Data engines?

Reply:
I was thinking many, but haven't thought about it sufficiently at this point...  Any preference?  Also: how / where to persist the state of last sync time?  (Should we just have part of the connector API be recording this last sync time back to the data engine via its connector?)

Reply:
Probably better to support many - it will affect the structur of the data engine proxy - whether you need a list of names + connections or just one.

The easiest implementation is to have a poll interval in the config and implement a separate thread per data engine proxy server instance to pull from each engine and then sleep for the poll interval and the call the data engines again.

There will also be data engines that have events - these will probably be integrated with a different type of connector (that is more like the event mapper).   Are you thinking that these should run in the same data engine proxy server implementation or a different type of server

Reply:
Probably the same -- meaning I'll need to add topic connectors at some point as well? -- but just trying to implement in small steps for now, and add that later 😄

Reply:
Raised a new issue to track this (for a future PR) under #1265